src/libCrypto/EthCrypto.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2022 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "EthCrypto.h"
+
+#include <openssl/ec.h>  // for EC_GROUP_new_by_curve_name, EC_GROUP_free, EC_KEY_new, EC_KEY_set_group, EC_KEY_generate_key, EC_KEY_free
+#include <openssl/obj_mac.h>  // for NID_secp192k1
+#include <openssl/sha.h>      //for SHA512_DIGEST_LENGTH
+#include <ethash/keccak.hpp>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+
+using namespace std;
+
+// Inspiration from:
+// https://stackoverflow.com/questions/10906524
+// https://stackoverflow.com/questions/57385412/
+// https://medium.com/mycrypto/the-magic-of-digital-signatures-on-ethereum-98fe184dc9c7
+
+// Prefix signed txs in Ethereum with Keccak256("\x19Ethereum Signed
+// Message:\n32" + Keccak256(message))
+constexpr uint8_t prelude[] = {25,  69,  116, 104, 101, 114, 101, 117, 109,
```

Comment:
I would suggest that we leave it as uint8_t because it contains non-ascii values which can cause issues around when there are x0 in it etc. Happy to move it into the one function